nixos/modules/services/web-apps/archtika.nix

```diff
@@ -0,0 +1,312 @@
+      recommendedZstdSettings = true;
+      recommendedOptimisation = true;
+
+      appendHttpConfig = ''
+        limit_req_zone $binary_remote_addr zone=requestLimit:10m rate=5r/s;
+        limit_req_status 429;
+        limit_req zone=requestLimit burst=20 nodelay;
+
+        add_header Strict-Transport-Security "max-age=31536000; includeSubDomains" always;
+        add_header X-Frame-Options "SAMEORIGIN" always;
+        add_header X-Content-Type-Options "nosniff" always;
+        add_header Referrer-Policy "strict-origin-when-cross-origin" always;
+        add_header Permissions-Policy "accelerometer=(),autoplay=(),camera=(),cross-origin-isolated=(),display-capture=(),encrypted-media=(),fullscreen=(self),geolocation=(),gyroscope=(),keyboard-map=(),magnetometer=(),microphone=(),midi=(),payment=(),picture-in-picture=(self),publickey-credentials-get=(),screen-wake-lock=(),sync-xhr=(self),usb=(),xr-spatial-tracking=(),clipboard-read=(self),clipboard-write=(self),gamepad=(),hid=(),idle-detection=(),interest-cohort=(),serial=(),unload=()" always;
+
+        map $http_cookie $auth_header {
+          default "";
+          "~*session_token=([^;]+)" "Bearer $1";
+        }
+      '';
+
+      virtualHosts = {
+        "${cfg.domain}" = {
+          useACMEHost = cfg.domain;
+          forceSSL = true;
+          locations = {
+            "/" = {
+              proxyPass = "http://localhost:${toString cfg.webAppPort}";
+            };
+            "/previews/" = {
+              alias = "/var/www/archtika-websites/previews/";
+              index = "index.html";
+              tryFiles = "$uri $uri/ $uri.html =404";
+            };
+            "/api/rpc/export_articles_zip" = {
+              proxyPass = "http://localhost:${toString cfg.apiPort}/rpc/export_articles_zip";
+              extraConfig = ''
+                default_type application/json;
+                proxy_set_header Authorization $auth_header;
+              '';
+            };
+            "/api/" = {
+              proxyPass = "http://localhost:${toString cfg.apiPort}/";
+              extraConfig = ''
+                default_type application/json;
+              '';
+            };
+            "/api/rpc/register" = mkIf cfg.settings.disableRegistration {
+              extraConfig = ''
+                deny all;
+              '';
+            };
+          };
+        };
+        "~^(?<subdomain>.+)\\.${cfg.domain}$" = {
+          useACMEHost = cfg.domain;
+          forceSSL = true;
+          locations = {
+            "/" = {
+              root = "/var/www/archtika-websites/$subdomain";
+              index = "index.html";
+              tryFiles = "$uri $uri/ $uri.html =404";
+            };
+          };
+        };
+      };
+    };
+
+    security.acme = {
+      acceptTerms = true;
+      defaults.email = cfg.acmeEmail;
+      certs."${cfg.domain}" = {
+        domain = cfg.domain;
+        extraDomainNames = [ "*.${cfg.domain}" ];
```

Comment:
This needs handling to skip setting these options if `cfg.acmeEmail` or `cfg.domain` is `null`.